AppConnectSwift/CaptureImage.swift

```diff
@@ -0,0 +1,137 @@
+    var image = UIImage()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        photoUpload(true)
+    }
+    
+    override func viewDidAppear(animated: Bool) {
+        super.viewDidAppear(animated)
+        self.navigationItem.title = "Capture Image"
+    }
+    
+    // MARK: Image picker delegate functions
+    func imagePickerController(picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : AnyObject]) {
+        self.dismissViewControllerAnimated(true, completion: nil)
+        print(info)
+        let image = info[UIImagePickerControllerOriginalImage] as? UIImage
+        imageView.image = image
+        self.image = image!
+    }
+    
+    func imagePickerControllerDidCancel(picker: UIImagePickerController) {
+        self.dismissViewControllerAnimated(true, completion: nil)
+    }
+    
+    // MARK: View Controller Toolbar Items
+    @IBAction func searchTapped(sender: AnyObject) {
+        photoUpload(false)
+    }
+    
+    @IBAction func cameraTapped(sender: AnyObject) {
+        photoUpload(true)
+    }
+    
+    @IBAction func saveTapped(sender: AnyObject) {
+        if(image.CGImage != nil){
+            let data = compressFile() as? NSData
+            var subject: MDSubject!
+            subject.collectData(data, withMetadata: "Random String", completion: { (dataEnvelop: MDSubjectDataEnvelope!, err: NSError!) -> Void in
+                if(err != nil){
+                    print(err?.description);
+                }
+                else{
+                    print("Data Saved");
+                }
+            })
+            // Empty the view once data successfully saved
+            imageView.image = nil
+        }
+        else{
+            postAlert("No image", message: "Image selected has no path")
+        }
+    }
+    
+    func photoUpload(fromCamera: Bool) {
+        imagePicker.allowsEditing = false
+        imagePicker.delegate = self
+        // Looks for camera
+        if(fromCamera){
+            if (UIImagePickerController.isSourceTypeAvailable(.Camera)) {
+                imagePicker.sourceType = .Camera
+                presentViewController(imagePicker, animated: true, completion: {})
+            } else {
+                postAlert("Camera not accessable", message: "AppConnect cannot access the camera.")
+            }
+        }
+        // Looks for images in photo library
+        else {
+            imagePicker.sourceType = .PhotoLibrary
+            presentViewController(imagePicker, animated: true, completion: {})
+        }
+    }
+
+    func postAlert(title: String, message: String) {
+        let alert = UIAlertController(title: title, message: message,
+        preferredStyle: UIAlertControllerStyle.Alert)
+        alert.addAction(UIAlertAction(title: "OK", style: UIAlertActionStyle.Default, handler: nil))
+        self.presentViewController(alert, animated: true, completion: nil)
+    }
+
+    func compressFile() -> NSData{
```

Comment:
It looks like this resizes the issue rather than compresses - maybe `resizeImage`?